torchao/quantization/sfdp_int8_fx_pass.py

```diff
@@ -0,0 +1,686 @@
+import functools
+from typing import Callable
+
+import torch
+from torch._inductor import config
+from torch._inductor.pattern_matcher import (
+    filter_nodes,
+    fwd_only,
+    register_replacement,
+    gen_register_replacement,
+    PatternMatcherPass,
+)
+from torch._dynamo.utils import counters
+from torch._inductor.fx_passes.fuse_attention import (
+    partialize_and_update_signature
+)
+from torchao.ops import scaled_dot_product_int8
+
+__all__ = [
+    "_sfdp_init_int8",
+]
+
+aten = torch.ops.aten
+patterns = PatternMatcherPass()
+
+def _sfdp_pattern_int8_1(
+    query,
+    key,
+    value,
+    attn_mask,
+    inv_scale,
+    q_zp,
+    q_scale,
+    k_zp,
+    k_scale,
+    v_zp,
+    v_scale,
+    a_zp,
+    a_scale,
+    o_zp,
+    o_scale,
+    dropout,
+):
+    # int8-mix-fp32 QUANTIZED SDPA with mask
+    q = query.permute([0, 2, 1, 3])
+    q = torch.ops.quantized_decomposed.dequantize_per_tensor.default(
```

Comment:
we are moving away from these ops, it would be better if you can use our newer quantize_affine/dequantize_affine ops directly from https://github.com/pytorch/ao/blob/main/torchao/quantization/quant_primitives.py